lbrynet/core/client/ConnectionManager.py

```diff
@@ -142,37 +142,21 @@ def _send_secondary_requests(self, peer, protocol):
         ]
         return defer.DeferredList(ds)
 
-    def protocol_disconnected(self, peer, protocol):
-        log.debug("%s protocol %s disconnected for %s",
-                    self._get_log_name(), type(protocol).__name__, peer)
-        if peer in self._peer_connections:
-            del self._peer_connections[peer]
-        if peer in self._connections_closing:
-            d = self._connections_closing[peer]
-            del self._connections_closing[peer]
-            d.callback(True)
-
-    def manage(self):
-        self._manage_deferred = self._manage()
-        if not self.stopped:
-            self._next_manage_call = self.callLater(self.MANAGE_CALL_INTERVAL_SEC, self.manage)
-
-
     @defer.inlineCallbacks
-    def _manage(self):
+    def manage(self, schedule_next_call=True):
+        self._manage_deferred = defer.Deferred()
         if len(self._peer_connections) < conf.settings['max_connections_per_stream']:
             log.debug("%s have %d connections, looking for %d",
                         self._get_log_name(), len(self._peer_connections),
                         conf.settings['max_connections_per_stream'])
             ordered_request_creators = self._rank_request_creator_connections()
             peers = yield self._get_new_peers(ordered_request_creators)
             peer = self._pick_best_peer(peers)
```

Comment:
remove the exception catching here, all exceptions should be caught and logged by ClientProtocolFactory.clientConnectionFailed()